libs/coin-framework/src/sanction/index.ts

```diff
@@ -0,0 +1,56 @@
+import { LiveConfig } from "@ledgerhq/live-config/LiveConfig";
+import axios from "axios";
+import { hours, makeLRUCache } from "@ledgerhq/live-network/cache";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { getEnv } from "@ledgerhq/live-env";
+
+const cache = getEnv("MOCK")
+  ? () => fetchSanctionedAddresses()
+  : makeLRUCache(fetchSanctionedAddresses, () => "all_sanctioned_addresses", hours(12));
+
+async function fetchSanctionedAddresses(): Promise<Record<string, string[]>> {
+  try {
+    const { data } = await axios.get(getEnv("SANCTIONED_ADDRESSES_URL"));
+    return data;
+  } catch (_) {
+    // We dont want if call fails for some reason to stop the workflow, user must be able to make transaction
+    return {};
+  }
+}
+
+export async function isAddressSanctioned(
+  currency: CryptoCurrency,
+  address: string,
+): Promise<boolean> {
+  if (!isCheckSanctionedAddressEnabled(currency)) {
+    return false;
+  }
+
+  // Only for testing, should be deleted after
+  const temporarySanctionedAddress: string[] | undefined = LiveConfig.getValueByKey(
+    "tmp_sanctioned_addresses",
+  );
+
+  const data: Record<string, string[]> = await cache();
+  const addresses = data["bannedAddresses"] || [];
+
+  if (temporarySanctionedAddress) {
+    return addresses.concat(temporarySanctionedAddress).includes(address);
```

Comment:
How long can "addresses" become? I assume we can't store it as a `set`?

Reply:
that's not too long so running `includes` on the array should be fine I guess - thx for the info